### PR TITLE
Implemented redux store with auth slice

### PR DESCRIPTION
**Pull request by Vinyl-Davyl**

### CODE REVIEW

**Comment by williamssam** [comment]
Hi @Vinyl-Davyl , Why did you switch off "noImplicitAny"??

File: client/src/store/slices/auth.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    reducers: {
+        login: (state, action) => {
+            state.user = action.payload
+            state.isLoggedIn = true
+            localStorage.setItem('teamwork-member', JSON.stringify(action.payload))
+        },
+        logout: (state) => {
+            state.user = null
+            state.isLoggedIn = false
+            localStorage.removeItem('team-menber')
```
Comment (pablo-clueless) [inline client/src/store/slices/auth.ts:41]: `localStorage` key should match with what we have on line 36

File: client/src/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (pablo-clueless) [inline client/src/types/index.ts:1]: Use the schema here, https://dbdiagram.io/d/63c79244296d97641d7a6111, to create the user type

**Review by pablo-clueless — CHANGES_REQUESTED** [review]

File: client/src/store/slices/auth.ts (hunk 1 above)
Comment (Vinyl-Davyl) [reply client/src/store/slices/auth.ts]: Yeah, Resolved typo

File: client/src/types/index.ts (hunk 2 above)
Comment (Vinyl-Davyl) [reply client/src/types/index.ts:1]: Okay

**Comment by Vinyl-Davyl** [comment]
> Hi @Vinyl-Davyl , Why did you switch off "noImplicitAny"?? and you also removed ErrorBoundary component from app.tsx??

Error boundary component?? Not sure about that. Didn't change any file there. But I switched off non Implicit any and some other stuff to resolve some type warning. Can always take the back though and find a way around

**Review by pablo-clueless — APPROVED** [review]

**Pull request merged**